### DISCUSSION TITLE
How can I use yup context on validations?

### DISCUSSION THREAD

**Discussion by Ridermansb** [mantinedev/mantine#3867] [Q&A] [answered]
Basically yup, in the validate options, has a `context` property that I can use to add more context on my validation

See more here: https://github.com/jquense/yup#schemavalidatevalue-any-options-object-promiseinfertypeschema-validationerror

How can I add a context to that?

**Comment by Ridermansb** [comment]
Hey.. anyone knows how to do this?

**Reply by thinkdj** [reply]
Did you try https://mantine.dev/form/validation/#yup ?

**Reply by Ridermansb** [reply]
This is only showing how I can integrate with yup.

But is not showing how I can use `context` .. 

Yup has this concept of context that I can use when I call for example [`validate`](https://github.com/jquense/yup#schemavalidatevalue-any-options-object-promiseinfertypeschema-validationerror) function so my schema has access to this context.

In react-hook-form for example, is passing by property in the form configuratino ([see here](https://react-hook-form.com/api/useform/#context))

But I don't know how I can do the same using manting/form

**Reply by cyantree** [reply, marked as answer]
I think you can provide your own `validate` method for `useForm()` that passes the context. The `yupResolver` is just a thin wrapper around `YupSchema.validateSync()`:
https://github.com/mantinedev/mantine/blob/master/src/mantine-form/src/resolvers/yup-resolver/yup-resolver.ts
